httpbp/admin.go

```diff
@@ -0,0 +1,21 @@
+package httpbp
+
+import (
+	"net/http"
+
+	"github.com/reddit/baseplate.go/internal/admin"
+	"github.com/reddit/baseplate.go/log"
+)
+
+// ServeAdmin starts a blocking HTTP server for internal functions:
+//    health        - serve /health for health checking
+//    metrics       - serve /metrics for prometheus
+//    profiling     - serve /debug/pprof for profiling, ref: https://pkg.go.dev/net/http/pprof
+//
+// Default server address is admin.Addr.
+//
+// This function blocks, so it should be run as its own goroutine.
+func ServeAdmin(healthCheck http.HandlerFunc) {
```

Comment:
if we move this into `httpbp`, things start to get trickier: we actually have [`httpbp.HandlerFunc`](https://pkg.go.dev/github.com/reddit/baseplate.go/httpbp#HandlerFunc) that's different from `http.HandlerFunc`.

I'm not saying we should just use `httpbp.HandlerFunc` here, but mixing `http.HandlerFunc` inside `httpbp` package can get really confusing.

Reply:
make it func() error?

Reply:
@fishy I don't have a preference what we use. You tell me whats best... either:
`func ServeAdmin(healthCheck http.HandlerFunc) {`, or
`func ServeAdmin(healthCheck httpbp.HandlerFunc) {`, or
what kyle is saying?

Reply:
I don't think `func() error` works for the case to support different health check probes. I would prefer `httpbp.HandlerFunc` to reduce confusion, but we need a good way to handle the error return (e.g. we need to write the error return to `ResponseWriter` in some way, which is not something straightforward, cc @pacejackson ).

but maybe for health check purpose we can just log the error and write a standard 500 error to `ResponseWriter` should be good enough?

Reply:
actually https://github.com/reddit/baseplate.go/blob/b1ba649a8929c56f48d57ed43074854082ce2c25/httpbp/handler.go#L34 is a good example on how to handle the error from `httpbp.HandlerFunc`